src/pspm_extract_segments_core.m

```diff
@@ -67,11 +67,13 @@
         onset = currentOnsets(j);
         endIndex = onset + segment_length - 1;
 
-        % Check if the segment extends beyond the data
+        % Check if the segment extends beyond the data. int64 avoids
+        % imprecision problems with previous rounding operations (which can
+        % in fact result in non-integer values)
         if endIndex > length(currentData)
-            segment = [currentData(onset:end), NaN(1, endIndex - length(currentData))];
+            segment = [currentData(int64(onset):end), NaN(1, endIndex - length(currentData))];
```

Comment:
Should the second `endIndex` be also `int64(endIndex)` too?

Reply:
This is now generalised.